### PR TITLE
add wasm_struct_obj_get_field_count API support.

### PR DESCRIPTION
**Pull request by LevelCA**

### CODE REVIEW

File: core/iwasm/common/gc/gc_object.c [hunk 1]
```diff
@@ -189,6 +189,20 @@ wasm_struct_obj_get_field(const WASMStructObjectRef struct_obj,
...
+{
+    uint32 field_count = 0;
+
+    WASMRttTypeRef rtt_type =
+        (WASMRttTypeRef)wasm_object_header((WASMObjectRef)struct_obj);
+    WASMStructType *struct_type = (WASMStructType *)rtt_type->defined_type;
+    if (struct_type->field_count > 0)
+        field_count = struct_type->field_count;
+
+    return field_count;
```
Comment (wenyongh) [inline core/iwasm/common/gc/gc_object.c:203]: How about `return struct_type->field_count` direclty and removing `uint32 field_count = 0`?

Comment (LevelCA) [reply core/iwasm/common/gc/gc_object.c]: OK done

File: core/iwasm/common/gc/gc_object.c [hunk 2]
```diff
@@ -189,6 +189,16 @@ wasm_struct_obj_get_field(const WASMStructObjectRef struct_obj,
     }
 }
 
+uint32
+wasm_struct_obj_get_field_count(const WASMStructObjectRef struct_obj)
+{
+    WASMRttTypeRef rtt_type =
+        (WASMRttTypeRef)wasm_object_header((WASMObjectRef)struct_obj);
+    WASMStructType *struct_type = (WASMStructType *)rtt_type->defined_type;
+    if (struct_type->field_count > 0)
```
Comment (xujuntwt95329) [inline core/iwasm/common/gc/gc_object.c:198]: Seems we don't need to add this check? Just return `struct_type->field_count`

And had better add an empty line after variable definition

Comment (LevelCA) [reply core/iwasm/common/gc/gc_object.c]: done

**Comment by wenyongh** [comment]
@LevelCA the `dev/gc_refactor` had been merged into `main` for some time, normally we don't need to maintain it again. Could you please rebase your code with main branch and submit the PR to main?

**Comment by LevelCA** [comment]
> @LevelCA the `dev/gc_refactor` had been merged into `main` for some time, normally we don't need to maintain it again. Could you please rebase your code with main branch and submit the PR to main?

OK, i will do it later.

**Pull request closed without merging**